broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/PartitionTransitionProcess.java

```diff
@@ -68,14 +70,16 @@ private void proceedWithTransition(final ActorFuture<Void> future) {
     concurrencyControl.run(
         () -> {
           final var nextStep = pendingSteps.remove(0);
-          startedSteps.push(nextStep);
 
           LOG.info(
               format("Transition to %s on term %d - executing %s", role, term, nextStep.getName()));
-
-          nextStep
-              .transitionTo(context, term, role)
-              .onComplete((ok, error) -> onStepCompletion(future, error));
+          try {
+            nextStep
+                .transitionTo(context, term, role)
+                .onComplete((ok, error) -> onStepCompletion(future, error));
+          } catch (final Exception | AssertionError e) {
+            future.completeExceptionally(e);
```

Comment:
@Zelldon In the other PR you asked whether this change is necessary (the bit about catching the exception),
It is necessary, because without it any exception thrown in `transitionTo` will be logged by the threads uncaught exception method and the transition will be stuck.

Reply:
For me it feels like we mix test with production code

Reply:
Because of the `AssertionError` ? Or how do you mean it?

Reply:
Yeah correct, also the try catch in general was only added because of the tests right? Before there was no try catch, in production code in this case the actor (zeebe partition)  would fail.

Reply:
Well, yes and no. It was discovered in tests that when an exception is thrown at this location, then the actor would fail. Which is a bug in my opinion. But happy to discuss this aspect. 

But the assertion error, yes, that is very specific to the tests and not needed.